src/sources/stdin.rs

```diff
@@ -40,32 +36,60 @@ impl SourceConfig for StdinConfig {
         _globals: &GlobalOptions,
         out: mpsc::Sender<Event>,
     ) -> crate::Result<super::Source> {
-        Ok(stdin_source(stdin(), self.clone(), out))
+        Ok(stdin_source(
+            io::BufReader::new(io::stdin()),
+            self.clone(),
+            out,
+        ))
     }
 
     fn output_type(&self) -> DataType {
         DataType::Log
     }
 }
 
-pub fn stdin_source<S>(stream: S, config: StdinConfig, out: mpsc::Sender<Event>) -> super::Source
+pub fn stdin_source<R>(stdin: R, config: StdinConfig, out: mpsc::Sender<Event>) -> super::Source
 where
-    S: AsyncRead + Send + 'static,
+    R: Send + io::BufRead + 'static,
 {
     Box::new(future::lazy(move || {
         info!("Capturing STDIN");
 
         let host_key = config.host_key.clone().unwrap_or(event::HOST.to_string());
         let hostname = hostname::get_hostname();
-
-        let source = FramedRead::new(
-            stream,
-            BytesDelimitedCodec::new_with_max_length(b'\n', config.max_length),
-        )
-        .map(move |line| create_event(line, &host_key, &hostname))
-        .map_err(|e| error!("error reading line: {:?}", e))
-        .forward(out.sink_map_err(|e| error!("Error sending in sink {}", e)))
-        .map(|_| info!("finished sending"));
+        let (mut tx, rx) = futures::sync::mpsc::channel(1024);
+
+        thread::spawn(move || {
+            let mut buf = Box::new(BytesMut::new());
```

Comment:
Not sure we need to box here?

Reply:
So we actually don't need a buffer because internally `lines()` does that. What we will want to do is use the `Bytes::from(<string>)` function to go from string -> bytes that we can send down the channel. Hopefully that makes sense :)

Reply:
Yes sure I am currently working on it